### PR TITLE
container_create: set cpuset cpus|mems

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

**- What I did**

set cpuset cpus|mems at container_create

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: server/container_create.go [hunk 1]
```diff
@@ -785,28 +785,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
...
-
-			memoryLimit := resources.MemoryLimitInBytes
-			if memoryLimit != 0 {
-				specgen.SetLinuxResourcesMemoryLimit(memoryLimit)
-			}
-
-			oomScoreAdj := resources.OomScoreAdj
-			specgen.SetProcessOOMScoreAdj(int(oomScoreAdj))
+			specgen.SetLinuxResourcesCPUPeriod(uint64(resources.GetCpuPeriod()))
+			specgen.SetLinuxResourcesCPUQuota(resources.GetCpuQuota())
```
Comment (TomSweeneyRedHat) [inline server/container_create.go]: Are there some tab vs space issues here and below at 79-794?  It looks OK in view mode however, so maybe a github preview hiccup.

File: server/container_create.go [hunk 2]
```diff
@@ -785,28 +785,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
...
+			specgen.SetLinuxResourcesCPUShares(uint64(resources.GetCpuShares()))
+			specgen.SetLinuxResourcesMemoryLimit(resources.GetMemoryLimitInBytes())
+			specgen.SetProcessOOMScoreAdj(int(resources.GetOomScoreAdj()))
+			specgen.SetLinuxResourcesCPUCpus(resources.GetCpusetCpus())
+			specgen.SetLinuxResourcesCPUMems(resources.GetCpusetMems())
```
Comment (TomSweeneyRedHat) [inline server/container_create.go]: so you don't need to set SetLinuxResousourcesCPUPeriod or CPUQuota anymore?

Comment (runcom) [reply server/container_create.go]: Look above, pretty sure I'm setting them

Comment (TomSweeneyRedHat) [reply server/container_create.go]: UGH, sorry runcom, was looking at just the ones below my other comment.  Time for more tea.  That looks fine to me.  Next time feel free to give me a harder head slap.

File: server/container_create.go [hunk 3]
```diff
@@ -785,28 +785,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
...
-			}
-
-			memoryLimit := resources.MemoryLimitInBytes
-			if memoryLimit != 0 {
-				specgen.SetLinuxResourcesMemoryLimit(memoryLimit)
-			}
-
-			oomScoreAdj := resources.OomScoreAdj
-			specgen.SetProcessOOMScoreAdj(int(oomScoreAdj))
+			specgen.SetLinuxResourcesCPUPeriod(uint64(resources.GetCpuPeriod()))
```
Comment (mrunalp) [inline server/container_create.go]: Are these values always set by the kubelet since we took out the 0 checks?

Comment (runcom) [reply server/container_create.go]: looks so by the check I did

Comment (mrunalp) [reply server/container_create.go]: okay!

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**